src/porepy/models/protocol.py

```diff
@@ -0,0 +1,825 @@
+"""Contains the protocols that declare the methods and attributes that must be present
+in an instance of a PorePy model. The proper implementations of these methods can be
+found in various classes within the models directiory.
+
+Note that the protocol framework is accessed by static type checkes only!
+
+Warning:
```

Comment:
Adding this warning to reflect the insights gained with @Yuriyzabegaev 
The runtime `PorePyModel` class does not inherit from `Protocol` anymore.